### ISSUE TITLE
Actionscript Tracking: Top Level

### ISSUE THREAD

**Issue by Dinnerbone** [ruffle-rs/ruffle#310]
This is the tracking issue for **Top Level** in AVM1. We will check off each item of progress as appropriate, and fill in any relevant or missing information as we continue development of Ruffle.
# Legend
Each checkbox is independent of another. It's entirely possible for something to be tested but not exist yet, or for us to believe that it's completely functional but we haven't made enough tests to prove it.
## "Property Exists"
This means the item exists, but may not necessarily be fully implemented.
## "Has Test Coverage"
This means that we have believe that we have a good test coverage of this item, regardless of if those tests pass. It's okay to have tests available but not implement the item yet.
## "Completely Functional"
This means we believe that the item is completely implemented, and no more work needs to be done towards making it functional.

# Top Level
### `_focusRect : Boolean`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `_global : Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `_highquality : Number`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `_quality : String`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `_root : MovieClip`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `_soundbuftime : Number`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `_forceframerate : Boolean`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `$version : Number`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `Infinity : Number`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `NaN : Number`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Accessibility`](https://github.com/ruffle-rs/ruffle/issues/239)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Array`](https://github.com/ruffle-rs/ruffle/issues/240)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`AsBroadcaster`](https://github.com/ruffle-rs/ruffle/issues/241)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Boolean`](https://github.com/ruffle-rs/ruffle/issues/242)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Button`](https://github.com/ruffle-rs/ruffle/issues/243)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Camera`](https://github.com/ruffle-rs/ruffle/issues/244)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Color`](https://github.com/ruffle-rs/ruffle/issues/245)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`ContextMenu`](https://github.com/ruffle-rs/ruffle/issues/246)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`ContextMenuItem`](https://github.com/ruffle-rs/ruffle/issues/247)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`CustomActions`](https://github.com/ruffle-rs/ruffle/issues/248)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Date`](https://github.com/ruffle-rs/ruffle/issues/249)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Enumeration`](https://github.com/ruffle-rs/ruffle/issues/250)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Error`](https://github.com/ruffle-rs/ruffle/issues/251)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`ExtendedKey`](https://github.com/ruffle-rs/ruffle/issues/252)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`File`](https://github.com/ruffle-rs/ruffle/issues/253)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`FontName`](https://github.com/ruffle-rs/ruffle/issues/254)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Function`](https://github.com/ruffle-rs/ruffle/issues/255)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`FunctionArguments`](https://github.com/ruffle-rs/ruffle/issues/256)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Key`](https://github.com/ruffle-rs/ruffle/issues/257)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`LoadVars`](https://github.com/ruffle-rs/ruffle/issues/258)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`LocalConnection`](https://github.com/ruffle-rs/ruffle/issues/259)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Math`](https://github.com/ruffle-rs/ruffle/issues/260)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Microphone`](https://github.com/ruffle-rs/ruffle/issues/261)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Mouse`](https://github.com/ruffle-rs/ruffle/issues/262)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`MovieClip`](https://github.com/ruffle-rs/ruffle/issues/263)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`MovieClipLoader`](https://github.com/ruffle-rs/ruffle/issues/264)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`NetConnection`](https://github.com/ruffle-rs/ruffle/issues/265)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`NetStream`](https://github.com/ruffle-rs/ruffle/issues/266)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Number`](https://github.com/ruffle-rs/ruffle/issues/267)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Object`](https://github.com/ruffle-rs/ruffle/issues/268)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`ObjectID`](https://github.com/ruffle-rs/ruffle/issues/269)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`PrintJob`](https://github.com/ruffle-rs/ruffle/issues/270)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Selection`](https://github.com/ruffle-rs/ruffle/issues/271)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`SharedObject`](https://github.com/ruffle-rs/ruffle/issues/272)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Sound`](https://github.com/ruffle-rs/ruffle/issues/273)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Stage`](https://github.com/ruffle-rs/ruffle/issues/274)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`String`](https://github.com/ruffle-rs/ruffle/issues/275)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`System`](https://github.com/ruffle-rs/ruffle/issues/276)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`System.IME`](https://github.com/ruffle-rs/ruffle/issues/277)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`System.capabilities`](https://github.com/ruffle-rs/ruffle/issues/278)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`System.security`](https://github.com/ruffle-rs/ruffle/issues/279)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`TextField`](https://github.com/ruffle-rs/ruffle/issues/280)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`TextField.StyleSheet`](https://github.com/ruffle-rs/ruffle/issues/281)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`TextFormat`](https://github.com/ruffle-rs/ruffle/issues/282)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`TextSnapshot`](https://github.com/ruffle-rs/ruffle/issues/283)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`URI`](https://github.com/ruffle-rs/ruffle/issues/284)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`Video`](https://github.com/ruffle-rs/ruffle/issues/285)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`XML`](https://github.com/ruffle-rs/ruffle/issues/286)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`XMLNode`](https://github.com/ruffle-rs/ruffle/issues/287)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`XMLSocket`](https://github.com/ruffle-rs/ruffle/issues/288)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`XMLUI`](https://github.com/ruffle-rs/ruffle/issues/289)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.display.BitmapData`](https://github.com/ruffle-rs/ruffle/issues/290)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.external.ExternalInterface`](https://github.com/ruffle-rs/ruffle/issues/291)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.BevelFilter`](https://github.com/ruffle-rs/ruffle/issues/292)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.BitmapFilter`](https://github.com/ruffle-rs/ruffle/issues/293)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.BlurFilter`](https://github.com/ruffle-rs/ruffle/issues/294)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.ColorMatrixFilter`](https://github.com/ruffle-rs/ruffle/issues/295)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.ConvolutionFilter`](https://github.com/ruffle-rs/ruffle/issues/296)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.DisplacementMapFilter`](https://github.com/ruffle-rs/ruffle/issues/297)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.DropShadowFilter`](https://github.com/ruffle-rs/ruffle/issues/298)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.GlowFilter`](https://github.com/ruffle-rs/ruffle/issues/299)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.GradientBevelFilter`](https://github.com/ruffle-rs/ruffle/issues/300)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.filters.GradientGlowFilter`](https://github.com/ruffle-rs/ruffle/issues/301)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.geom.ColorTransform`](https://github.com/ruffle-rs/ruffle/issues/302)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.geom.Matrix`](https://github.com/ruffle-rs/ruffle/issues/303)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.geom.Point`](https://github.com/ruffle-rs/ruffle/issues/304)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.geom.Rectangle`](https://github.com/ruffle-rs/ruffle/issues/305)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.geom.Transform`](https://github.com/ruffle-rs/ruffle/issues/306)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.net.FileReference`](https://github.com/ruffle-rs/ruffle/issues/307)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.net.FileReferenceList`](https://github.com/ruffle-rs/ruffle/issues/308)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### [`flash.text.TextRenderer`](https://github.com/ruffle-rs/ruffle/issues/309)`: Object`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional

### `call(frame)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `clearInterval(id)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `clearTimeout(id)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `duplicateMovieClip(target, newName, depth)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `escape(value)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `eval(expr)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `fscommand(command, parameters)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `getProperty(movieClip, property)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `getTimer()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `getURL(url, window, method)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `getVersion()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `gotoAndPlay(sceneOrFrame, frame)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `gotoAndStop()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `int(num)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `isFinite(value)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `isNaN(value)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `loadMovie(url, target, method)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `loadMovieNum(url, level, method)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `loadVariables(url, target, method)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `loadVariablesNum(url, level, method)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `nextFrame()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `nextScene()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `parseFloat(value)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `parseInt(value, radix)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `play()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `prevFrame()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `prevScene()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `profile(enable)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `print(target, boundingBox)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `printAsBitmap(target, boundingBox)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `printAsBitmapNum(level, boundingBox)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `printNum(level, boundingBox)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `random(num)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `removeMovieClip(target)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `setInterval()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `setProperty(movieClip, property, value)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `setTimeout(functionReference, delay, args)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `showRedrawRegions(enable)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `startDrag(target, lockCenter, left, top, right, bottom)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `stop()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `stopAllSounds()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `stopDrag()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `targetPath(mc)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `toggleHighQuality()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `trace(value)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `unescape(value)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `unloadMovie(target)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `unloadMovieNum(level)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `updateAfterEvent()`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional
### `MMExecute(expr)`
- [ ] Property Exists
- [ ] Has Test Coverage
- [ ] Completely Functional

**Comment by Herschel** [comment]
Here is some content that requires `setTimeout`/`setInterval` along with `updateAfterEvent` for the preloader: https://www.newgrounds.com/portal/view/558833

Currently gets stuck on the preloader because those functions aren't implemented.
(Ironically this is code from some official Newgrounds preloaders I programmed eons ago... !)

**Comment by nagayev** [comment]
@Dinnerbone Hello.
Is this issue corresponds to the current state of affairs?
We already have implemented parseFloat as I know.

**Comment by Juanepaktho** [comment]
> This is the tracking issue for **Top Level** in AVM1. We will check off each item of progress as appropriate, and fill in any relevant or missing information as we continue development of Ruffle.
> 
> # Legend
> 
> Each checkbox is independent of another. It's entirely possible for something to be tested but not exist yet, or for us to believe that it's completely functional but we haven't made enough tests to prove it.
> 
> ## "Property Exists"
> 
> This means the item exists, but may not necessarily be fully implemented.
> 
> ## "Has Test Coverage"
> 
> This means that we believe that we have a good test coverage of this item, regardless of if those tests pass. It's okay to have tests available but not implement the item yet.
> 
> ## "Completely Functional"
> 
> This means we believe that the item is completely implemented, and no more work needs to be done towards making it functional.
> 
> 
> 
> # Top Level
> 
> ## Properties
> 
> ### `_focusRect : Boolean`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `_global : Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `_highquality : Number`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `_quality : String`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `_root : MovieClip`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `_soundbuftime : Number`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `$version : Number`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `Infinity : Number`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `NaN : Number`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Accessibility`](https://github.com/ruffle-rs/ruffle/issues/239)`: Object`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Array`](https://github.com/ruffle-rs/ruffle/issues/240)`: Object`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`AsBroadcaster`](https://github.com/ruffle-rs/ruffle/issues/241)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`Boolean`](https://github.com/ruffle-rs/ruffle/issues/242)`: Object`
> 
> - [X] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`Button`](https://github.com/ruffle-rs/ruffle/issues/243)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Camera`](https://github.com/ruffle-rs/ruffle/issues/244)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Color`](https://github.com/ruffle-rs/ruffle/issues/245)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`ContextMenu`](https://github.com/ruffle-rs/ruffle/issues/246)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`ContextMenuItem`](https://github.com/ruffle-rs/ruffle/issues/247)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`CustomActions`](https://github.com/ruffle-rs/ruffle/issues/248)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Date`](https://github.com/ruffle-rs/ruffle/issues/249)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`Error`](https://github.com/ruffle-rs/ruffle/issues/251)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`Function`](https://github.com/ruffle-rs/ruffle/issues/255)`: Object`
> 
> - [X] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`FunctionArguments`](https://github.com/ruffle-rs/ruffle/issues/256)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Key`](https://github.com/ruffle-rs/ruffle/issues/257)`: Object`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`LoadVars`](https://github.com/ruffle-rs/ruffle/issues/258)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`LocalConnection`](https://github.com/ruffle-rs/ruffle/issues/259)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Math`](https://github.com/ruffle-rs/ruffle/issues/260)`: Object`
> 
> - [X] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`Microphone`](https://github.com/ruffle-rs/ruffle/issues/261)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Mouse`](https://github.com/ruffle-rs/ruffle/issues/262)`: Object`
> 
> - [X] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`MovieClip`](https://github.com/ruffle-rs/ruffle/issues/263)`: Object`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`MovieClipLoader`](https://github.com/ruffle-rs/ruffle/issues/264)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`NetConnection`](https://github.com/ruffle-rs/ruffle/issues/265)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`NetStream`](https://github.com/ruffle-rs/ruffle/issues/266)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Number`](https://github.com/ruffle-rs/ruffle/issues/267)`: Object`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Object`](https://github.com/ruffle-rs/ruffle/issues/268)`: Object`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`PrintJob`](https://github.com/ruffle-rs/ruffle/issues/270)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Selection`](https://github.com/ruffle-rs/ruffle/issues/271)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`SharedObject`](https://github.com/ruffle-rs/ruffle/issues/272)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Sound`](https://github.com/ruffle-rs/ruffle/issues/273)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Stage`](https://github.com/ruffle-rs/ruffle/issues/274)`: Object`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`String`](https://github.com/ruffle-rs/ruffle/issues/275)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`System`](https://github.com/ruffle-rs/ruffle/issues/276)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`TextField`](https://github.com/ruffle-rs/ruffle/issues/280)`: Object`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`TextFormat`](https://github.com/ruffle-rs/ruffle/issues/282)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`TextSnapshot`](https://github.com/ruffle-rs/ruffle/issues/283)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`Video`](https://github.com/ruffle-rs/ruffle/issues/285)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`XML`](https://github.com/ruffle-rs/ruffle/issues/286)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`XMLNode`](https://github.com/ruffle-rs/ruffle/issues/287)`: Object`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`XMLSocket`](https://github.com/ruffle-rs/ruffle/issues/288)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.display.BitmapData`](https://github.com/ruffle-rs/ruffle/issues/290)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.external.ExternalInterface`](https://github.com/ruffle-rs/ruffle/issues/291)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`flash.filters.BevelFilter`](https://github.com/ruffle-rs/ruffle/issues/292)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.filters.BitmapFilter`](https://github.com/ruffle-rs/ruffle/issues/293)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`flash.filters.BlurFilter`](https://github.com/ruffle-rs/ruffle/issues/294)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`flash.filters.ColorMatrixFilter`](https://github.com/ruffle-rs/ruffle/issues/295)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.filters.ConvolutionFilter`](https://github.com/ruffle-rs/ruffle/issues/296)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.filters.DisplacementMapFilter`](https://github.com/ruffle-rs/ruffle/issues/297)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.filters.DropShadowFilter`](https://github.com/ruffle-rs/ruffle/issues/298)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.filters.GlowFilter`](https://github.com/ruffle-rs/ruffle/issues/299)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.filters.GradientBevelFilter`](https://github.com/ruffle-rs/ruffle/issues/300)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.filters.GradientGlowFilter`](https://github.com/ruffle-rs/ruffle/issues/301)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.geom.ColorTransform`](https://github.com/ruffle-rs/ruffle/issues/302)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`flash.geom.Matrix`](https://github.com/ruffle-rs/ruffle/issues/303)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.geom.Point`](https://github.com/ruffle-rs/ruffle/issues/304)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`flash.geom.Rectangle`](https://github.com/ruffle-rs/ruffle/issues/305)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.geom.Transform`](https://github.com/ruffle-rs/ruffle/issues/306)`: Object`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### [`flash.net.FileReference`](https://github.com/ruffle-rs/ruffle/issues/307)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.net.FileReferenceList`](https://github.com/ruffle-rs/ruffle/issues/308)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### [`flash.text.TextRenderer`](https://github.com/ruffle-rs/ruffle/issues/309)`: Object`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> 
> 
> ## Methods
> 
> ### `ASSetPropFlags(object, properties, setFlags, clearFlags)`
> 
> - [X] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `clearInterval(id)`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `clearTimeout(id)`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `duplicateMovieClip(target, newName, depth)`
> 
> (this compiles to `Action::CloneSprite`)
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `escape(value)`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `getURL(url, window, method)`
> 
> (this compiles to `Action::GetURL` or `Action::GetURL2`)
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `gotoAndPlay(sceneOrFrame, frame)`
> 
> (this compiles to `Action::GotoFrame` or `Action::GotoFrame2`)
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `gotoAndStop()`
> 
> (this compiles to `Action::GotoFrame` or `Action::GotoFrame2`)
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `isFinite(value)`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `isNaN(value)`
> 
> - [X] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `loadMovie(url, target, method)`
> 
> (this compiles to `Action::GetURL2`)
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `loadVariables(url, target, method)`
> 
> (this compiles to `Action::GetURL2`)
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `nextFrame()`
> 
> (this compiles to `Action::NextFrame`)
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `parseFloat(value)`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `parseInt(value, radix)`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `play()`
> 
> (this compiles to `Action::Play`)
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `prevFrame()`
> 
> (this compiles to `Action::PreviousFrame`)
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `removeMovieClip(target)`
> 
> (this compiles to `Action::RemoveSprite`)
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `setInterval()`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `setTimeout(functionReference, delay, args)`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `showRedrawRegions(enable)`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `startDrag(target, lockCenter, left, top, right, bottom)`
> 
> (this compiles to `Action::StartDrag`)
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `stop()`
> 
> (this compiles to `Action::Stop`)
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `stopDrag()`
> 
> (this compiles to `Action::EndDrag`)
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `trace(value)`
> 
> - [x] Property Exists
> 
> - [x] Has Test Coverage
> 
> - [x] Completely Functional
> 
> ### `unescape(value)`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `unloadMovie(target)`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `updateAfterEvent()`
> 
> - [x] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> ### `MMExecute(expr)`
> 
> - [ ] Property Exists
> 
> - [ ] Has Test Coverage
> 
> - [ ] Completely Functional
> 
> 
> 
> 
> 
>

**Comment by ROBERT-MCDOWELL** [comment]
is there a page where new developers can see the current status of this tracking and eventually get involved easily by clicking on a link going to each method/propery/function sources?

**Comment by gamalhetari** [comment]
Issue with the game Cyber Chaser, when bosses die it does not end level.

Not sure if this is the right place to post this problem,

**Comment by kumaravel-ruf** [comment]
How to load xml file in Ruffle

**Comment by Toad06** [comment]
There is also a `print` function/method (and compiling to `Action::GetURL2`) currently missing in the main post. See #6154 for an example.

**Comment by danielhjacobs** [comment]
There's also [printAsBitmap](https://open-flash.github.io/mirrors/as2-language-reference/global_functions.html#printAsBitmap()), [printAsBitmapNum](https://open-flash.github.io/mirrors/as2-language-reference/global_functions.html#printAsBitmapNum()), and [printAsNum](https://open-flash.github.io/mirrors/as2-language-reference/global_functions.html#printAsBitmapNum()).

**Comment by unblocked-games-school** [comment]
update please.

**Comment by ROBERT-MCDOWELL** [comment]
is anyone can review and update this roadmap?

**Comment by aspdfihaeuf** [comment]
> is anyone can review and update this tracking?

been waiting for an update fr

**Comment by ErdincBARUT** [comment]
This project will be super if you complete successfully like original flash player. I'm following with excitement. If there is no performance problem in mobile, I think, that will pack a punch to  2D/UI game industry. I hope you will be more fast. 
good works.

**Comment by ROBERT-MCDOWELL** [comment]
@ErdincBARUT well, up to you and others to contribute to the project to go faster. The flash to rust conversion is huge and more we will be, faster it wil be...

**Comment by TurkPashaDede** [comment]
Can't you hurry up and finish it faster?

**Comment by minhngoc25a** [comment]
@TurkPashaDede, it's astonishing how entitled you sound. You're using free software created by volunteers in their spare time, yet you have the audacity to demand instant service and berate them without an ounce of gratitude. It's about time someone called you out on your disrespectful behavior. Understand this clearly: your entitled attitude won't get you anywhere. These volunteers owe you nothing. If you want their help, start showing some basic decency and respect. Otherwise, be prepared to be ignored and left to your own devices. Respect is earned, and it's high time you learned that lesson.

**Comment by ROBERT-MCDOWELL** [comment]
@TurkPashaDede
whey not learn rust code nd help the developers community? more we are more it will go fast.

**Comment by Nameless-GitHubber** [comment]
@TurkPashaDede, please be patient! Adobe Flash is home to **thousands** of classes, methods, and APIs created over the years. Thus, it'll take some time before everything is implemented. For now, just enjoy Ruffle for what it is or learn the Rust programming language to help the community finish the project more quickly. You can also play your favorite Flash games using the standalone Adobe Flash Player (preferably with the [Flashpoint Archive](https://flashpointarchive.org)).

**Comment by MatejMagat305** [comment]
> Can't you hurry up and finish it faster?

how you dare you ..., it is amazing project, which is made by people as "free time hobby", if you want to faster, you can sponzor them ...

**Comment by ROBERT-MCDOWELL** [comment]
btw I encourage the main developers to provide some rust coding easy example of how the implemented X function or else to help people coming here to achieve the last part of the roadmap... thanks

**Comment by TurkPashaDede** [comment]
How much longer before Ruffle is ready?

**Comment by Dinnerbone** [comment]
> How much longer before Ruffle is ready?

Ready for what, exactly?

We're already compatible with more content than not.

**Comment by Hexstream** [comment]
> Ready for what, exactly?

Ready for version 1.0, "normal" releases rather than nightlies, and around 100% compatibility, I assume...

_Keep up the great work!_ 💪

**Comment by christopher3242** [comment]
more keep!

**Comment by milosPavic** [comment]
Excellent Job, just keep going :)

**Comment by ROBERT-MCDOWELL** [comment]
A nice project from an ex adobe RTMP RTMFP expert
https://github.com/zenomt/rtwebsocket

**Comment by Enturktainment7** [comment]
Will this be ever finishd or why is it only stuck at 95 percent on their page? 

It cant play flash videos for some reaosn, just shows black screen with nothing. Very annoying.

Hope people can be faster.

**Comment by Hexstream** [comment]
People are working as fast as possible, there is a ***LOT*** of "unseen" work in a massive project like this.

Progress is being made every day and [they are working towards version 1.0](https://ruffle.rs/blog/2024/09/12/optimisations-text-more).

**Comment by ROBERT-MCDOWELL** [comment]
EXCELLENT!

**Comment by ROBERT-MCDOWELL** [comment]
@Enturktainment7 
the dev kept what is the most difificult to convert, all netwok/stream/GPU/audio/video/data related....

**Comment by Nameless-GitHubber** [comment]
> Will this be ever finishd or why is it only stuck at 95 percent on their page?
> 
> It cant play flash videos for some reaosn, just shows black screen with nothing. Very annoying.
> 
> Hope people can be faster.

The team is making all the contributions it can, and emulating Flash very accurately takes a lot of time and effort. If you'd like, you can learn Rust yourself to speed up the development process or play your favorite Flash games using the [Flashpoint Archive](https://flashpointarchive.org).

**Comment by Enturktainment7** [comment]
Im not complaining. Im just boosting you people.

**Comment by ROBERT-MCDOWELL** [comment]
@Enturktainment7 read your comment again, and don't take people as stupid here....

**Comment by Hexstream** [comment]
_Can we block this troll already?_ o_o

**Comment by Enturktainment7** [comment]
You're a bunch of horrendous assholes.

**Comment by ROBERT-MCDOWELL** [comment]
@Enturktainment7 your parents certainly had a lot of issues with you..... I'm sad for them.

**Comment by Nameless-GitHubber** [comment]
This is truly an SMH moment. 🤦‍♂️

**Comment by yvvki** [comment]
The language support says 99% for AVM 1 compatibility, what's the remaining 1%?

**Comment by ROBERT-MCDOWELL** [comment]
check above the unchecked checkbox!

**Comment by rikxard** [comment]
Hello! is it possible to load additional swfs above the current one?

**Comment by Hexstream** [comment]
@rikxard I'm not sure what you mean, but this probably belongs in a [new issue](https://github.com/ruffle-rs/ruffle/issues)?

**Comment by rikxard** [comment]
Hello!

Thank you for your quick and kind reply. I guess it'll be a new issue as
you mention. I`m new to Ruffle trying to revive an old nice Flash website.
What I'm trying to do is that the ground level swf loads another swf over
it. I'll open the issue anyway.

Thank you again!

Ricardo

**Comment by LeoKids** [comment]
> [@Enturktainment7](https://github.com/Enturktainment7) the dev kept what is the most difificult to convert, all netwok/stream/GPU/audio/video/data related.... so be tolerant and try to help rather than to complain

Asking someone to be faster is not complaining or spam,It's him trying to boost people to be faster,It's not intended to be a complaint like you're saying....

**Comment by ROBERT-MCDOWELL** [comment]
@LeoKids depends if your mother tong is English or else. but saying this to an open source project where developers are mostly volunteers is considered as disrespectful and deserve the opposite of the expecting result.
Rather. if this guy had a little consciousness about how an open source works so he won't say anything like a leacher can do and will learn how he can contribute to the project to go faster.

**Comment by LeoKids** [comment]
> [@LeoKids](https://github.com/LeoKids) depends if your mother tong is English or else. but saying this to an open source project where developers are mostly volunteers is considered as disrespectful and deserve the opposite of the expecting result. Rather. if this guy had a little consciousness about how an open source works so he won't say anything like a leacher can do and will learn how he can contribute to the project to go faster.

My mother toungue isn't english,but it's portuguese,but i understand what you're trying to say....

**Comment by yvvki** [comment]
> <deleted> source code that barrowxu have when he worked at macromedia

Whoa, is that even legal? Though it would be helpful as a reference.

**Comment by Dinnerbone** [comment]
Please do not post links to leaked source code (real or not), this project will not use it.

**Comment by ROBERT-MCDOWELL** [comment]
anyhow I forked it and save it locally. when a gov becomes illegal since decades so everything is illegal....
adobe ruined my company abandoning flash for the web, and we are tens of thousands in my situation. so this is at least a refund for all the time I passed to develop in actionscript during my life.

**Comment by ROBERT-MCDOWELL** [comment]
> > source code that barrowxu have when he worked at macromedia
> 
> Whoa, is that even legal? Though it would be helpful as a reference.

indeed. it can be very helpful to speedup the dev, avoiding any guess and waste of time. meaning we can use it as a doc reference ....

**Comment by adrian17** [comment]
Let me repeat what Dinnerbone said, we’re not going to use that repo or even look into its contents. We only use sources that were cleared legally, and that currently is only the avmplus repo.

(Also, this is extremely offtopic for this specific issue)

**Comment by danielhjacobs** [comment]
This thread has been mostly off-topic, so I'm going to lock this to collaborators to keep it focused on updates.

For bug reports, [please don't hesitate to file a new issue](https://github.com/ruffle-rs/ruffle/issues).
For discussion or questions, [you can visit our discussion page](https://github.com/ruffle-rs/ruffle/discussions) or [Discord](https://discord.gg/ruffle).

Thank you!
